simple_certmanager/models.py

```diff
@@ -10,12 +10,53 @@
 )
 from privates.fields import PrivateMediaFileField
 
+from simple_certmanager.csr_generation import generate_csr_and_private_key
+
 from .constants import CertificateTypes
 from .mixins import DeleteFileFieldFilesMixin
 from .utils import load_pem_x509_private_key, pretty_print_certificate_components
 from .validators import PrivateKeyValidator, PublicCertValidator
 
 
+class SigningRequest(models.Model):
+    common_name = models.CharField(
+        max_length=100,
+        help_text=_("The common name for the certificate"),
+    )
+    country_name = models.CharField(
+        max_length=2, help_text=_("Two-letter country code"), blank=True, default="NL"
+    )
+    organization_name = models.CharField(
+        max_length=100, help_text=_("The name of the organization"), blank=True
+    )
+    state_or_province_name = models.CharField(
+        max_length=100, help_text=_("The state or province name"), blank=True
+    )
+    email_address = models.EmailField(
+        help_text=_("Email address for the certificate"), blank=True
+    )
+    csr = models.TextField(
+        editable=False,
+        blank=True,
+        help_text=_("Certificate Signing Request"),
+        verbose_name=_("CSR"),
+    )
+    private_key = models.TextField(
+        editable=False,
+        blank=True,
+    )
+
+    def save(self, *args, **kwargs):
+        generate_csr_and_private_key(self)
+        super().save(*args, **kwargs)
+
+    def __str__(self):
+        return _("Signing Request #{pk} for {common_name}").format(
+            pk=self.pk,
+            common_name=self.common_name,
+        )
```

Comment:
we follow django's contributing guidelines to structure models, so the `save` method should be defined after the `__str__`` magic method (see https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#model-style)